experiments/quicksilver/experiment.py

```diff
@@ -59,9 +59,10 @@ def compute_spack_section(self):
         system_specs["compiler"] = "default-compiler"
         system_specs["mpi"] = "default-mpi"
 
-        # empty package_specs value implies external package
-        self.add_spack_spec(system_specs["mpi"])
+        if pkg_manager == "spack":
+            # empty package_specs value implies external package
+            self.add_spack_spec(system_specs["mpi"])
 
-        self.add_spack_spec(
-            self.name, [f"quicksilver@{app_version} +mpi", system_specs["compiler"]]
-        )
+            self.add_spack_spec(
+                self.name, [f"quicksilver@{app_version} +mpi", system_specs["compiler"]]
+            )
```

Comment:
It looks like all of these individual subclasses of `Experiment` could be left alone, and you could skip `compute_spack_section` in `compute_package_section_wrapper` if `package_manager` is not spack. Am I missing something? It looks like all these functions are turned effectively into noops when the package manager is not spack, so it seems like it would make sense to skip the function altogether.

Reply:
Good point. Upon checking, the `environment-modules` manager in ramble actually does need this information, to create the file where modules are loaded. I removed the if-statements in the subclasses [313ff73](https://github.com/LLNL/benchpark/pull/616/commits/313ff7364f7b3041aacd8398b380c7ea28b25d4f). Having the `package_specs` be empty for non-spack managers was a mistake.